docs/proposals/devfile.md

```diff
@@ -0,0 +1,64 @@
+For [springboot petstore example](https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/java-web-spring/devfile.yaml) it would look like this: `odo push --run-command="maven build" --build-command="run webapp"`.
+This will be only temporary measure until we can mark commands directly in devfile.
+    * The best approach would probably be to use “fat” pods. Component will consist of a single Pod typically with two containers - “build” container and “run” container. Both containers will share a volume with the source code (emptyDir or PVC?)
+    * **Build container** will use the image (dockerImage) based on a component field in build command definition in devfile. The main command should be something that just sleeps indefinitely.  Only ad-hoc commands (build command) will be executed in this container.
+    * **Run container** will use the image  (dockerImage) based on a component field in run command definition in devfile. The main command will be the process that can control other processes. We could reuse Go implementation of the SupervisorD as it is currently used in odo s2i flow. \
+The SupervisorD will have one service configured. This service will run “run” command based on a definition in devfile. Thanks to the SupervisorD approach, we don’t need to keep connection to the cluster in order to have the application up and running
+    * The `odo push` flow when using the devfile.yaml
+        1. Create “fat” pod with run and build container as defined above, if it already doesn’t exist.
+        2. Sync files from local directory to $CHE_PROJECTS_ROOT (/projects by default) directory in build container. It actually doesn’t matter which container is used, as both should have shared volume mounted in this location
+        3. Execute build command in build pod and wait for it to finish
+        4. Invoke command that will restart the run service in run container. (supervisord ctl restart run)
+
+### Extending Devfile format
+
+Work with folks from Che side to extend Devfile format with additional information that is required by odo.
+
+* Add a command (like build, run)
+* Url/Ingres/Routes support
+* Storage support
+* Linking multiple components (applications)
+* Bootstrapping services from OperatorHub and linking it to the component (application)
+
+### Implement odo operations on top of a Devfile
+
+* Implement commands that will modify Devfile.yaml. Uses shouldn’t need to touch Devfile.yaml, everything should be handled by odo commands \
+For example:
+  * odo create url - instead of modifing .odo/config.yaml it will add all necessary information into the Devfile.yaml
+  * odo delete url - will delete url information from Devfile.yaml
```

Comment:
Nitpick but for the sake of correctness, it's `odo url craete` and `odo url delete`.

Reply:
fixed